src/analysis/retail/evoker/devastation/modules/AplCheck/index.tsx

```diff
@@ -20,64 +21,421 @@ const hasEssenceRequirement = (resources: number) => {
 };
 
 const COMMON_TOP: Rule[] = [
+  // Spend Snapfire procs ASAP
+  {
+    spell: TALENTS.FIRESTORM_TALENT,
+    condition: cnd.buffPresent(SPELLS.SNAPFIRE_BUFF),
+  },
+  {
+    spell: TALENTS.SHATTERING_STAR_TALENT,
+    condition: cnd.describe(cnd.optionalRule(cnd.spellAvailable(TALENTS.DRAGONRAGE_TALENT)), () => (
+      <>
+        you are about to cast <SpellLink spell={TALENTS.DRAGONRAGE_TALENT} /> <strong>(AoE)</strong>
+      </>
+    )),
+  },
+  {
+    spell: TALENTS.DRAGONRAGE_TALENT,
+    condition: cnd.describe(
+      cnd.optionalRule(
+        cnd.or(
+          cnd.targetsHit(
+            { atLeast: 3 },
+            { lookahead: 3000, targetType: EventType.Damage, targetSpell: SPELLS.PYRE },
+          ),
+          cnd.and(
+            cnd.spellCooldownRemaining(SPELLS.FIRE_BREATH, { atMost: 4000 }),
```

Comment:
is this the reverse logic of "dont send FB/ES if DR has <13s cd"? can send DR if FB has 4s and ES has 10s?